scipy/stats/_axis_nan_policy.py

```diff
@@ -31,30 +31,19 @@ def _broadcast_array_shapes(arrays, axis=None):
     return _broadcast_shapes(shapes, axis)
 
 
-def _broadcast_shapes(shapes, axis=None):
+def _process_axis(axis, shapes):
     """
-    Broadcast shapes, ignoring incompatibility of specified axes
+    Validate, remove negative elements, and sort axis tuple
     """
-    if not shapes:
-        return shapes
-
-    # input validation
-    if axis is not None:
+    if (axis is not None) and shapes:
```

Comment:
Just a thought: if we don't like special cases for `axis` `None`, we could turn that into `axis = np.arange(n_dims)` here.

Reply:
It would! I tried it locally but it gave me some non-trivial errors. Maybe, tackle that in another PR?

Reply:
OK. Let's remember to come back to this if you think there is something wrong.